packages/wallet/src/PersonalWallet/PersonalWallet.ts

```diff
@@ -137,19 +138,32 @@ export const DEFAULT_LOOK_AHEAD_SEARCH = 20;
 // Ideally we should calculate this based on the activeSlotsCoeff and probability of a single block per epoch.
 const BLOCK_SLOT_GAP_MULTIPLIER = 2.5;
 
-const isOutgoingTx = (input: Cardano.Tx | TxCBOR | OutgoingTx): input is OutgoingTx =>
+const getSerializedTxCBOR = (input: TxCBOR) => TxCBOR.deserialize(input).body;
+const isOutgoingTx = (input: Cardano.Tx | TxCBOR | OutgoingTx | SubmitTxArgs): input is OutgoingTx =>
   typeof input === 'object' && 'cbor' in input;
-const isTxCBOR = (input: Cardano.Tx | TxCBOR | OutgoingTx): input is TxCBOR => typeof input === 'string';
-const processOutgoingTx = (input: Cardano.Tx | TxCBOR | OutgoingTx): OutgoingTx => {
+const isTxCBOR = (input: Cardano.Tx | TxCBOR | OutgoingTx | SubmitTxArgs): input is TxCBOR => typeof input === 'string';
+const isSubmitTxArgs = (input: Cardano.Tx | TxCBOR | OutgoingTx | SubmitTxArgs): input is SubmitTxArgs =>
+  typeof input === 'object' && 'context' in input;
+const processOutgoingTx = (input: Cardano.Tx | TxCBOR | OutgoingTx | SubmitTxArgs): OutgoingTx => {
   // TxCbor
   if (isTxCBOR(input)) {
     return {
-      body: TxCBOR.deserialize(input).body,
+      body: getSerializedTxCBOR(input),
       cbor: input,
       // Do not re-serialize transaction body to compute transaction id
       id: Cardano.TransactionId.fromTxBodyCbor(TxBodyCBOR.fromTxCBOR(input))
     };
   }
+  // SubmitTxArgs
+  if (isSubmitTxArgs(input)) {
+    return {
+      body: input.signedTransaction.body,
+      cbor: TxCBOR.serialize(input.signedTransaction),
```

Comment:
Let's refactor `SubmitTxArgs` to `{ cbor: TxCBOR }` as we have 2 types that represent the same thing. Delete the old `SerializedTransaction` type that was used there. Please do that in a separate commit.
```suggestion
      cbor: input.signedTransaction,
```

Reply:
While trying this refactor, I found that interestingly this breaks quite a few things. The code for the `SubmitTxArgs` has `signedTransaction` set to `SerializedTransaction` which for me is resolving to type `any` because there is no `HexBlob` type in the current import path. As a result there seem to be some places that are doing unexpected things with the `signedTransaction` assuming its actually an object. 
I think this does need to be refactored, but I think that it should be it's own ticket after having a look at it. 

Perhaps it would be better to untangle this in a separate ticket?